### ISSUE TITLE
Question about the first OverlapPatchEmbed

### ISSUE THREAD

**Issue by pieteroyens** [NVlabs/SegFormer#59]
First of all, great work done by you and you colleagues, the results are really impressive. 

For my own application and understanding, I'm doing a deep dive through all the operations and I am encountering the following problem with my understanding:

In your paper you state that: "Given an image of size H × W × 3, we first divide it into patches of size 4 × 4."

Relating that back to your code, I believe that this image division is done within the first patch_embedding layer: 

`
self.patch_embed1 = OverlapPatchEmbed(img_size=img_size, patch_size=7, stride=4, in_chans=in_chans,
                                              embed_dim=embed_dims[0])
`

However, I find it strange that the img_size is set to 224 for whichever data we use. 

`
class OverlapPatchEmbed(nn.Module):
  def __init__(self, img_size=224, patch_size=7, stride=4, in_chans=3, embed_dim=768)
`

The numbers all workout, for example, with the ADE dataset and input images of size 512x512 we expect (512 / 4)^2 = 16384 patches, and we get back that amount after the first patch_embedding. 

But the image_size=224 seems arbitrary other than to make the numbers work. This is also in combination with the patch_size=7, which seems to contradict what is said in the paper. 

I am probably misunderstanding something and I would just like to clear this up, many thanks.

**Comment by jeya-maria-jose** [comment]
I also had the same question. 

I am also not sure where the 4x4 patch dividing is done. From my understading, the OverlapPatchEmbed is just a convlayer. The img_size argument is actually not used anywhere within the OverlapPatchEmbed function so img_size=224 is not an issue. So, I believe the conv function emulates the patch merging as we flatten and normalize the output of conv layer across that dimension. 

The question of where images are divided into 4x4 still remains!

**Comment by Waao666** [comment]
You can see the paper.

it's state that: "Overlapped Patch Merging. Given an image patch, the patch merging process used in ViT, unifies a N × N × 3 patch into a 1 × 1 × C vector. This can easily be extended to unify a 2 × 2 × Ci feature path into a 1 × 1 × Ci+1 vector to obtain hierarchical feature maps. Using this, we can shrink our hierarchical features from F1 ( H4 × W4 × C1) to F2 ( H8 × W8 × C2), and then iterate for any other feature map in the hierarchy. This process was initially designed to combine non-overlapping image or feature patches. Therefore, it fails to preserve the local continuity around those patches. Instead, we use an overlapping patch merging process. To this end, we define K, S, and P, where K is the patch size, S is the stride between two adjacent patches, and P is the padding size. In our experiments, we set K = 7, S = 4, P = 3 ,and K = 3, S = 2, P = 1 to perform overlapping patch merging to produces features with the same size as the non-overlapping process."

So, I guess the author's meaning is, one patch contain 4*4 new pixels information, and rest 3(7-4) is overlapped pixels.
